src/plugins/sync-data-attrs/host.js

```diff
@@ -0,0 +1,6 @@
+module.exports = function hostSyncDataAttrs(host) {
+	host.onRequest('attrs', function() {
+		var htmlElems = document.getElementsByTagName('html');
+		if (htmlElems.length === 1 && htmlElems[0].hasAttributes()) return Object.assign({}, htmlElems[0].dataset);
```

Comment:
Nit: might be easier to return `{}` if there's no HTML element or no attributes on it -- that way the request will always return _something_.